pkg/datapath/linux/node_ids.go

```diff
@@ -25,24 +25,19 @@ const (
 
 // AllocateNodeID allocates a new ID for the given node (by IP) if one wasn't
 // already assigned.
+// This should never happen. If it ever does, we may have an unexpected
+// call to AllocateNodeID.
+// Don't allocate a node ID for the local node.
 func (n *linuxNodeHandler) AllocateNodeID(nodeIP net.IP) uint16 {
 	if len(nodeIP) == 0 || nodeIP.IsUnspecified() {
-		// This should never happen. If it ever does, we may have an unexpected
```

Comment:
should this comment stay here?

Reply:
@jibi good catch - yes it should stay here - thanks. seems to be reordered when moving this check (&warning) back to `AllocateNodeID` because it doesn't make sense to have this in `getNodeIDForIP`

Reply:
moved the comments back to its original place